### PR TITLE
Feature #502 - Remove error handler and check previous vote

### LINKED ISSUE #502: Implement Voting Logic
Make use of `governor.buildVoteTx()` and  `governor.signAndPost()` to implement the voting mechanism.

### PR DESCRIPTION
**Pull request by abefernan**
**Description**

Makes some fixes regarding the Voting Panel:
- Removes the error handler from the Voting Panel now that #544 is implemented
- Adds a `previousVote` state so that if a user clicks on the same current vote it isn't cast
- Changes the ocurrencies of `creationDate`s (which were specified in the design wireframes) to `startDates` (which better represent the actual data).

Closes #502.

### CODE REVIEW

File: packages/sil-voting-app/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ interface Props {
 
 const Buttons = ({ id, vote }: Props): JSX.Element => {
   const [currentVote, setCurrentVote] = useState(vote);
+  const [previousVote, setPreviousVote] = useState(currentVote);
```
Comment (webmaster128) [inline packages/sil-voting-app/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx]: This could also be `= useState(currentVote);` like above, right? If you do that, it makes the source of the initial state more direct and you can swap the order of those lines easily

Comment (abefernan) [reply packages/sil-voting-app/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx]: Do you mean `const [previousVote, setPreviousVote] = useState(vote);`? If not I'm afraid I don't understand.

Comment (webmaster128) [reply packages/sil-voting-app/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx]: I do

File: packages/sil-voting-app/src/store/proposals/actions.ts [hunk 2]
```diff
@@ -43,12 +43,14 @@ export async function getProposals(governor: Governor): Promise<ProposalsState>
   const proposals = await governor.getProposals();
   const proposalsState = await Promise.all(
     proposals.map(async proposal => {
+      const creationDate = new Date(proposal.votingStartTime * 1000);
```
Comment (webmaster128) [inline packages/sil-voting-app/src/store/proposals/actions.ts]: `creationDate` -> `startDate`

File: packages/sil-voting-app/src/store/proposals/actions.ts [hunk 3]
```diff
@@ -61,6 +63,7 @@ export async function getProposals(governor: Governor): Promise<ProposalsState>
         },
         result: proposal.result,
         vote: await getVote(proposal),
+        hasStarted: creationDate < new Date(),
```
Comment (webmaster128) [inline packages/sil-voting-app/src/store/proposals/actions.ts]: There is no operator overloading in JavaScript. So I think `<` between `Date` objects does not what you think it does. You can use `.getTime()` on both sides and compare milliseconds as integers

File: packages/sil-voting-app/src/store/proposals/dummyData.ts [hunk 4]
```diff
@@ -217,12 +217,14 @@ export const getDummyProposalsState = (): ProposalsState => {
   const getVote = (proposal: Proposal): VoteOption | undefined => getDummyVote(proposal);
 
   const proposalsState = proposals.map(proposal => {
+    const creationDate = new Date(proposal.votingStartTime * 1000);
```
Comment (webmaster128) [inline packages/sil-voting-app/src/store/proposals/dummyData.ts]: `startDate`

File: packages/sil-voting-app/src/store/proposals/dummyData.ts [hunk 5]
```diff
@@ -235,6 +237,7 @@ export const getDummyProposalsState = (): ProposalsState => {
       },
       result: proposal.result,
       vote: getVote(proposal),
+      hasStarted: creationDate < new Date(),
```
Comment (webmaster128) [inline packages/sil-voting-app/src/store/proposals/dummyData.ts]: as above

File: packages/sil-voting-app/src/store/proposals/actions.ts (hunk 3 above)
Comment (willclarktech) [reply packages/sil-voting-app/src/store/proposals/actions.ts]: A couple of StackOverflow answers I've seen suggest this actually works fine. (You can use `<` and `>`, but not `==`.)

https://stackoverflow.com/a/13285569

Comment (webmaster128) [reply packages/sil-voting-app/src/store/proposals/actions.ts]: Alright, 👍 then

**Comment by webmaster128** [comment]
Not critical but something to keep in mind: here we compare a local time with an on-chain time. This can potentially cause troube when the machine's time is out of sync and thus the user cannot vote even if the voting time did not end on chain. It would be possible to get the time from the chain (via block headers). But for MVP this is unnecessarily complicated.

File: packages/sil-governance/src/routes/dashboard/components/Proposal/Period.tsx [hunk 6]
```diff
@@ -3,22 +3,22 @@ import React from "react";
...
+  const startLocaleDate = startDate.toLocaleString();
 
   return (
     <Block display="flex" alignItems="center">
       <Typography variant="body2" weight="semibold">
         {expiryLabel} {expiryLocaleDate}
       </Typography>
       <Block marginLeft={1}>
-        <Typography variant="body2">Created on {creationLocaleDate}</Typography>
+        <Typography variant="body2">Created on {startLocaleDate}</Typography>
```
Comment (webmaster128) [inline packages/sil-governance/src/routes/dashboard/components/Proposal/Period.tsx]: Do I understand correctly this is the date when the voting on the proposal starts? Then the proposal can be created a long time before the voting actually starts. So I suggest the texts "Starts on" / "Started on" like we have with `expiryLabel`.

Sorry if this is out of scope for this PR. You can also create a different ticket if you want to merge as is

Comment (abefernan) [reply packages/sil-governance/src/routes/dashboard/components/Proposal/Period.tsx]: Yes, the proper creation date of the proposal doesn't belong in the UI, I believe. The wireframes called the "start date" "creation date". I'll update the label no problem.

Comment (webmaster128) [reply packages/sil-governance/src/routes/dashboard/components/Proposal/Period.tsx]: Creation date of the proposal is not important for now. But the governor needs to know when they can vote.
